examples/jsm/nodes/functions/PhysicalLightingModel.js

```diff
@@ -28,14 +32,53 @@ const computeMultiscattering = ( singleScatter, multiScatter, specularF90 = floa
 
 };
 
-const RE_IndirectSpecular_Physical = new ShaderNode( ( inputs ) => {
+const LM_Init = new ShaderNode( ( context, stack, builder ) => {
+
+	if ( builder.includes( clearcoat ) ) {
```

Comment:
Isn't this a bit of a hack?

Reply:
I'm trying to convince myself that yes, but I think that currently is the better approach. I created a class to contextualize `LighingModel` but it creates more classes that can replaced by `.includes()`, basically if the user does not declare/assign `clearcoat` in `.constructVariants()` we can use `builder.includes( clearcoat )` to detect that `clearcoat` is not used in this material.